utils/tracking.ts

```diff
@@ -0,0 +1,59 @@
+import { ServicesPreferencesModeEnum } from "@pagopa/io-functions-commons/dist/generated/definitions/ServicesPreferencesMode";
+import { RetrievedProfile } from "@pagopa/io-functions-commons/dist/src/models/profile";
+import { initTelemetryClient } from "./appinsights";
+import { toHash } from "./crypto";
+
+export const createTracker = (
+  telemetryClient: ReturnType<typeof initTelemetryClient>
+) => {
+  const eventName = (name: string) => `api.profile.${name}`;
+
+  /**
+   * Trace an event when a user changes their preference mode
+   */
+  const traceServicePreferenceModeChange = (
+    hashedFiscalCode: string,
+    previousMode: ServicesPreferencesModeEnum,
+    nextMode: ServicesPreferencesModeEnum
+  ) =>
+    telemetryClient.trackEvent({
+      name: eventName("change-service-preferences-mode"),
+      properties: {
+        nextMode,
+        previousMode,
+        userId: hashedFiscalCode
+      },
+      tagOverrides: { samplingEnabled: "false" }
+    });
+
+  /**
+   * Trace an event when a user has previous preferences to migrate
+   */
+  const traceMigratingServicePreferences = (
+    oldProfile: RetrievedProfile,
+    newProfile: RetrievedProfile,
+    action: "REQUESTING" | "DOING"
```

Comment:
Why not a `DONE` event?

Reply:
Good point, done in https://github.com/pagopa/io-functions-app/pull/185/commits/3f99d9c65d3cff73d11f22a6e97a7702d2035a19